api/src/main/kotlin/com/google/devtools/ksp/processing/Resolver.kt

```diff
@@ -56,6 +56,24 @@ interface Resolver {
      */
     fun getClassDeclarationByName(name: KSName): KSClassDeclaration?
 
+    /**
+     * Find a property in the compilation classpath for the given name.
+     *
+     * @param name fully qualified name of the function to be loaded; using '.' as separator.
+     * @param includeTopLevel a boolean value indicate if top level functions should be searched. Default false. Note if top level functions are included, this operation can be expensive.
+     * @return a KSFunctionDeclaration, or null if not found.
+     */
+    fun getFunctionDeclarationByName(name: KSName, includeTopLevel: Boolean = false): KSFunctionDeclaration?
```

Comment:
There can be several functions with exactly the same name. I.e., overloading.

Reply:
This is handled by the implementation in `getAllFunctions()`

Reply:
When there are two functions with the same name, this will return null. Shouldn't it return a list?

Reply:
Oh you are right, I was thinking about overriding.